### PR TITLE
Expanded upon Google Analytics event API

### PR DESCRIPTION
**Pull request by nalbion**
https://developers.google.com/analytics/devguides/collection/analyticsjs/command-queue-reference#send

### CODE REVIEW

**Comment by horiuchi** [comment]
@nalbion Thank you for your contribution.
Please fix Travis CI error: https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/102481354

**Comment by nalbion** [comment]
hi @horiuchi I've made a few updates, but I'm afraid I don't understand the remaining errors

File: google.analytics/ga.d.ts [hunk 1]
```diff
@@ -38,13 +38,46 @@ interface GoogleAnalytics {
...
+            socialNetwork: string, socialAction: string, socialTarget: string): void;
+        (command: 'send', hitType: 'social',
+            fieldsObject: {socialNetwork: string, socialAction: string, socialTarget: string}): void;
+        (command: 'send', hitType: 'timing',
+            timingCategory: string, timingVar: string, timingValue: number): void;
+        (command: 'send', hitType: 'timing',
+            fieldsObject: {timingCategory: string, timingVar: string, timingValue: number}): void;
+        (command: 'send', hitType: HitType, ...fields: any): void;
+        (command: 'send', fieldsObject: {}): void;
+
```
Comment (horiuchi) [inline google.analytics/ga.d.ts]: `(command: string, hitType: string, ...fields: any[]): void;`
We need this line for non-specialized signature.

**Comment by nalbion** [comment]
What's this last error about?  
    Specialized overload signature is not assignable to any non-specialized signature.

File: google.analytics/ga.d.ts [hunk 2]
```diff
@@ -38,13 +38,46 @@ interface GoogleAnalytics {
...
+            socialNetwork: string, socialAction: string, socialTarget: string): void;
+        (command: 'send', hitType: 'social',
+            fieldsObject: {socialNetwork: string, socialAction: string, socialTarget: string}): void;
+        (command: 'send', hitType: 'timing',
+            timingCategory: string, timingVar: string, timingValue: number): void;
+        (command: 'send', hitType: 'timing',
+            fieldsObject: {timingCategory: string, timingVar: string, timingValue: number}): void;
+        (command: 'send', hitType: HitType, ...fields: any[]): void;
+        (command: 'send', fieldsObject: {}): void;
+
```
Comment (horiuchi) [inline google.analytics/ga.d.ts]: `(command: string, hitType: string, ...fields: any[]): void;`
We need this line for non-specialized signature.

File: google.analytics/ga.d.ts [hunk 3]
```diff
@@ -38,13 +38,47 @@ interface GoogleAnalytics {
...
+        (command: 'send', hitType: 'social',
+            fieldsObject: {socialNetwork: string, socialAction: string, socialTarget: string}): void;
+        (command: 'send', hitType: 'timing',
+            timingCategory: string, timingVar: string, timingValue: number): void;
+        (command: 'send', hitType: 'timing',
+            fieldsObject: {timingCategory: string, timingVar: string, timingValue: number}): void;
+        (command: 'send', hitType: HitType, ...fields: any[]): void;
+        (command: 'send', fieldsObject: {}): void;
+        
+        (command: string, hitType: string, ...fields: any[]): void;
```
Comment (nalbion) [inline google.analytics/ga.d.ts]: restored, as requested

Comment (horiuchi) [inline google.analytics/ga.d.ts]: That's good. Your commented error is corrected.

**Comment by nalbion** [comment]
woohoo!

**Comment by horiuchi** [comment]
@nalbion Thanks mate! nice work :+1:

**Pull request merged**